nucypher/utilities/networking.py

```diff
@@ -200,4 +204,62 @@ def determine_external_ip_address(
     # complete failure!
     if not rest_host:
         raise UnknownIPAddress('External IP address detection failed')
+
     return rest_host
+
+
+def _is_global_ipv4(ip: str) -> bool:
+    """Check if an IP address is a global IPv4 address according to RFC 1918."""
+    try:
+        ip = ip_address(ip.strip())
+        return isinstance(ip, IPv4Address) and ip.is_global
+    except AddressValueError:
+        return False
+
+
+def _resolve_ipv4(ip: str) -> Optional[str]:
+    """Resolve an IP address to IPv4 if required and possible."""
+    try:
+        ip = ip_address(ip.strip())
+    except AddressValueError:
+        return None
+    if isinstance(ip, IPv6Address) and ip.ipv4_mapped:
+        return str(ip.ipv4_mapped)
+    elif isinstance(ip, IPv4Address):
+        return str(ip)
+
+
+def _extract_ip(header: str, request: Request) -> Optional[str]:
+    """Extract the first IP address from a request header."""
+    if header in request.headers:
+        for ip in request.headers[header].split(","):
+            yield ip
+
+
+def _ip_sources(request: Request) -> str:
+    """Iterate over all possible sources of IP addresses in a request's headers."""
+    for header in ["X-Forwarded-For", "X-Real-IP"]:
```

Comment:
How did we select those headers? Are these industry standards for proxies/forwarding?